src/architecture/inline-layout.md

```diff
@@ -0,0 +1,263 @@
+- [wpt/css/CSS2/bidi-text](https://wpt.fyi/results/css/CSS2/bidi-text?label=master&product=chrome%5Bexperimental%5D&product=edge%5Bexperimental%5D&product=firefox%5Bexperimental%5D&product=safari%5Bexperimental%5D&product=servo&aligned)
+- [css/CSS2/text/bidi-flag-emoji](https://wpt.fyi/results/css/CSS2/text/bidi-flag-emoji.html?label=master&product=chrome%5Bexperimental%5D&product=edge%5Bexperimental%5D&product=firefox%5Bexperimental%5D&product=safari%5Bexperimental%5D&product=servo&aligned)
+- [css/CSS2/text/bidi-span-003](https://wpt.fyi/results/css/CSS2/text/bidi-span-003.html?label=master&product=chrome%5Bexperimental%5D&product=edge%5Bexperimental%5D&product=firefox%5Bexperimental%5D&product=safari%5Bexperimental%5D&product=servo&aligned)
+- ...
+
+#### Font Style matching procedure
+Exact details is provided in servo [fonts module](./fonts.md)
+
+### Text Segmentation & Shaping
+#### What is text shaping?
+Wikipedia provides following definition:
+> Text shaping is the process of converting text to glyph indices and positions as part of text rendering. It is complementary to font rendering as part of the text rendering process; font rendering is used to generate the glyphs, and text shaping decides which glyphs to render and where they should be put on the image plane. Unicode is generally used to specify the text to be rendered.
+
+Microsoft have the following document that describe [Text layout](https://learn.microsoft.com/en-us/globalization/fonts-layout/text-layout). It have section devoted to [Text shaping](https://learn.microsoft.com/en-us/globalization/fonts-layout/text-layout#text-shaping), however clear and short definition is not provided, this section names four important subtasks that any text shping engine must solve:
+- [correct processing of ligatures](https://learn.microsoft.com/en-us/globalization/fonts-layout/text-layout#ligatures)
+- [script-specific replacement of the characters that depends on context](https://learn.microsoft.com/en-us/globalization/fonts-layout/text-layout#contextual-shaping)
+- [combining special characters (i.e diacritics and tone marks) into single visual representation](https://learn.microsoft.com/en-us/globalization/fonts-layout/text-layout#combining-characters)
+- [script-specific (i.e Hindi, Devanagari) reordering of the characters](https://learn.microsoft.com/en-us/globalization/fonts-layout/text-layout#character-reordering)
+
+[HarfBuzz definition](https://harfbuzz.github.io/what-is-harfbuzz.html) of text shaping:
+> Text shaping is the process of translating a string of character codes (such as Unicode codepoints) into a properly arranged sequence of glyphs that can be rendered onto a screen or into final output form for inclusion in a document.
```

Comment:
this information is quite generic, so it’s probably enough to link to those pages without quoting them.